addons/stock/models/stock_picking.py

```diff
@@ -413,7 +413,7 @@ def _compute_show_check_availability(self):
                 float_compare(move.product_uom_qty, 0, precision_rounding=move.product_uom.rounding)
                 for move in picking.move_lines
             )
-            picking.show_check_availability = picking.is_locked and picking.state in ('confirmed', 'waiting', 'assigned') and has_moves_to_reserve
+            picking.show_check_availability = (picking.is_locked and picking.state in ('confirmed', 'waiting', 'assigned') and has_moves_to_reserve) and not picking.immediate_transfer
```

Comment:
it may be more readable to do the immediate transfer check earlier in the loop
```
for picking in self:
    if picking.immediate_transfer:
        picking.show_check_availability = False
        continue
    has_moves_to_reserve....
```

Reply:
and even more readable if all the conditions to fail are done earlier
```
    def _compute_show_check_availability(self):
        """ According to `picking.show_check_availability`, the "check availability" button will be
        displayed in the form view of a picking.
        """
        for picking in self:
            if picking.immediate_transfer or\
                    not picking.is_locked or\
                    picking.state not in ('confirmed', 'waiting', 'assigned'):
                picking.show_check_availability = False
                continue
            picking.show_check_availability = any(
                move.state in ('waiting', 'confirmed', 'partially_available') and
                float_compare(move.product_uom_qty, 0, precision_rounding=move.product_uom.rounding)
                for move in picking.move_lines
            )
```